training_system/app/controllers/trainer/user_courses_controller.rb

```diff
@@ -0,0 +1,66 @@
+class Trainer::UserCoursesController < ApplicationController
+  before_action :load_course, only: :create
+  before_action :load_course_destroy, only: :destroy
+
+  def new; end
+
+  def edit; end
+
+  def create
+    if params[:user_ids].present?
+      assign_trainee
+    else
+      flash[:danger] = t "controllers.course_controller.assign_trainee.f"
+      redirect_to edit_trainer_course_path @course.id
+    end
+  end
+
+  def destroy
+    trainees = User.user_in_course params[:user_ids]
+    if @course_destroy && trainees
+      @course_destroy.transaction do
+        @course_destroy.users.destroy trainees
+      end
+      flash[:success] = t "controllers.course_controller.remove_trainee.success"
+      redirect_to edit_trainer_course_path @course_destroy.id
+    else
+      flash[:danger] = t "controllers.course_controller.remove_trainee.fail"
+      redirect_to edit_trainer_course_path @course.id
+    end
+  end
+
+  private
+
+  def load_course
+    @course = Course.find_by id: params[:course_id]
+    return if @course
+
+    flash.now[:danger] = t "controllers.course_controller.error_show"
+    render :edit
+  end
+
+  def load_course_destroy
+    @course_destroy = Course.find_by id: params[:id]
+    return if @course_destroy
+
+    flash.now[:danger] = t "controllers.course_controller.error_show"
+    render :edit
+  end
+
+  def assign_trainee
+    @course.transaction do
+      params[:user_ids].each do |user|
+        @course.user_courses.build user_id: user
+      end
+      if @course.save!(validate: false)
```

Comment:
`(validate: false)` Sao lại disable validate của xử lý này thế

Reply:
do trong model course có `validates :course_subjects, presence: true` mà khi em save thì nó validate luôn chỗ này báo là `course subjects cant be blank` nên e muốn disable validate này lại nhưng mà disable 1 cái này thì em tìm không ra nên em disable validate khi save ạ

Reply:
chỗ này em nên comment code `validates :course_subjects, presence: true`  vì sau này cần code này nhé
chứ em skip validate thì nó đâu có ý nghĩa, sẽ ko bao giờ rollback dc